pandas/io/parsers.py

```diff
@@ -749,13 +753,24 @@ def read(self, nrows=None):
         ret = self._engine.read(nrows)
 
         if self.options.get('as_recarray'):
+            # In this case we are not updating self._currow... but we do not
+            # need it
             return ret
 
         # May alter columns / col_dict
         index, columns, col_dict = self._create_index(ret)
 
+        if index is None:
+            # Any column is actually fine:
+            new_rows = len(col_dict[columns[0]])
+            index = RangeIndex(self._currow, self._currow + new_rows)
+        else:
+            new_rows = len(index)
```

Comment:
use ``len(col_dict.values()[0])`` as I think this might fail with duplicate column names (add a test for that as well)